From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tariq Bashir <120014322+t-bashir-bs@users.noreply.github.com>
Date: Mon, 15 Jan 2024 15:48:16 +0000
Subject: OS-16546: enable file cookies

Allow websites from file:// domain to access cookies
Cherry picked from QtWebEngine I638d2cd24ea426d78cdaa90a58b94559bab4588f

diff --git a/net/cookies/cookie_monster.cc b/net/cookies/cookie_monster.cc
index e1090c36a6fcba47216df837d777880c64d41a98..129e03a90aaa7b450754d83b4ccff4505610f4df 100644
--- a/net/cookies/cookie_monster.cc
+++ b/net/cookies/cookie_monster.cc
@@ -681,7 +681,7 @@ void CookieMonster::SetPersistSessionCookies(bool persist_session_cookies) {
 
 // static
 std::vector<std::string> CookieMonster::GetDefaultCookieableSchemes() {
-  return std::vector<std::string>{"http", "https", "ws", "wss"};
+  return std::vector<std::string>{"http", "https", "ws", "wss", "file"};
 }
 
 CookieChangeDispatcher& CookieMonster::GetChangeDispatcher() {
